crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -26,7 +26,16 @@ class ApplicationController < ActionController::Base
   rescue_from Crowbar::Error::ChefOffline, with: :chef_is_offline
 
   before_filter :enable_profiler, if: proc { ENV["ENABLE_PROFILER"] == "true" }
-  before_filter :enforce_installer, unless: proc { Crowbar::Installer.successful? || ENV["RAILS_ENV"] == "test" }
+  before_filter :enforce_installer, unless: proc {
+    Crowbar::Installer.successful? || \
+    ENV["RAILS_ENV"] == "test" || \
+    Rails.cache.read(:sanity_check_errors).any?
+  }
+  before_filter :sanity_checks_cache
+  before_filter :sanity_checks, unless: proc {
```

Comment:
Also, I'm not totally sure why we need two filters here; I'd put `sanity_checks_cache` inside `sanity_checks`.